scikeras/utils/_transformers.py

```diff
@@ -0,0 +1,262 @@
+            self.n_classes_ = self.classes_.size
+        elif self.target_type == "multiclass-one-hot":
+            self.classes_ = np.arange(0, y.shape[1])
+            self.n_classes_ = y.shape[1]
+        self.n_outputs_ = 1
+        self.model_n_outputs_ = 1
+        self.y_dtype_ = y.dtype
+        return self
+
+    def transform(self, X: np.ndarray) -> np.ndarray:
+        y = X  # rename for clarity, the input is always expected to be a target `y`
+        # TODO: validate self.classes_ and self.n_classes_, n_outputs_, model_n_outputs_
+        return self.final_encoder_.transform(y)
+
+    def inverse_transform(
+        self, X: np.ndarray, return_proba: bool = False
+    ) -> np.ndarray:
+        y = X  # rename for clarity, the input is always expected to be a target `y`
+        if self.target_type == "binary":
+            # array([0.9, 0.1], [.2, .8]) -> array(['yes', 'no'])
+            if self.n_classes_ == 1:
+                # special case: single input label for sigmoid output
+                # may give more predicted classes than inputs for
+                # small sample sizes!
+                # don't even bother inverse transforming, just fill.
+                class_predictions = np.full(
+                    shape=(y.shape[0], 1), fill_value=self.classes_[0]
+                )
+            else:
+                if y.shape == 1 or (y.shape[1] == 1 and self.n_classes_ == 2):
+                    # result from a single sigmoid output
+                    # reformat so that we have 2 columns
+                    y = np.column_stack([1 - y, y])
+                y_ = np.argmax(y, axis=1).reshape(-1, 1)
+                class_predictions = self.final_encoder_.inverse_transform(y_)
+        elif self.target_type == "multiclass":
+            # array([0.8, 0.1, 0.1], [.1, .8, .1]) ->
+            # array(['apple', 'orange'])
+            idx = np.argmax(y, axis=-1)
+            if not is_categorical_crossentropy(self.loss):
+                y_ = idx.reshape(-1, 1)
+            else:
+                y_ = np.zeros(y.shape, dtype=int)
+                y_[np.arange(y.shape[0]), idx] = 1
+            class_predictions = self.final_encoder_.inverse_transform(y_)
+        else:  # "multiclass-one-hot"
+            # array([0.8, 0.1, 0.1], [.1, .8, .1]) ->
+            # array([[1, 0, 0], [0, 1, 0]])
+            idx = np.argmax(y, axis=-1)
+            y_ = np.zeros(y.shape, dtype=int)
+            y_[np.arange(y.shape[0]), idx] = 1
+            class_predictions = y_
+
+        if return_proba:
+            return y
+        else:
+            return np.squeeze(np.column_stack(class_predictions)).astype(
+                self.y_dtype_, copy=False
+            )
+
+    def get_meta_params(self):
+        return {
+            "classes_": self.classes_,
+            "n_classes_": self.n_classes_,
+            "n_outputs_": self.n_outputs_,
+            "model_n_outputs_": self.model_n_outputs_,
+        }
```

Comment:
User gets to decide what attributes they want to return for setting in the main SciKeras wrapper